cumulus/polkadot-omni-node/lib/src/command.rs

```diff
@@ -99,8 +110,34 @@ fn new_node_spec(
 }
 
 /// Parse command line arguments into service configuration.
-pub fn run<CliConfig: crate::cli::CliConfig>(cmd_config: RunConfig) -> Result<()> {
-	let mut cli = Cli::<CliConfig>::from_args();
+/// Entrypoint to run the CLI with optional extra subcommands.
+///
+/// This function wraps the parsing of arguments and dispatching to either:
+/// - A built-in subcommand (e.g., `build-spec`, `export-genesis-head`)
+/// - A user-defined extra subcommand via the `ExtraSubcommand` trait
+///
+/// # Type Parameters
+/// - `CliConfig`: Runtime-specific configuration (e.g., network, spec versions)
+/// - `Extra`: One or more subcommands implementing `ExtraSubcommand`
+pub fn run<CliConfig, Extra: ExtraSubcommand>(cmd_config: RunConfig) -> Result<()>
```

Comment:
This should help with omitting the second generic type if no extra subcommand is wanted. Might need an import too in the file.

```suggestion
pub fn run<CliConfig, Extra: ExtraSubcommand = NoExtraSubcommand>(cmd_config: RunConfig) -> Result<()>
```